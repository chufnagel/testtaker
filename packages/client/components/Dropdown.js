import PropTypes from "prop-types";

const Dropdown = ({ action, choices, name }) => (
  <div className="dropdown-container">
    <select onChange={action} name={name} className="dropdown">
      {choices.map((choice, idx) => (
        <option key={choice} value={choice}>
          {choice}
        </option>
      ))}
    </select>
  </div>
);

Dropdown.propTypes = {
  choices: PropTypes.array,
};

Dropdown.defaultProps = {
  choices: ["Latin"],
};

export default Dropdown;
