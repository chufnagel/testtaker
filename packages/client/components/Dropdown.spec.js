/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import Dropdown from "./Dropdown";

// Dropdown.propTypes = {
//   action: PropTypes.func.isRequired,
//   choices: PropTypes.arrayOf(PropTypes.string),
//   name: PropTypes.string,
// };

// Dropdown.defaultProps = {
//   choices: ["Latin"],
//   name: "Lorem Ipsum",
// };

const mockFn = jest.fn();

describe("Dropdown", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Dropdown action={mockFn} />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("Dropdown matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
