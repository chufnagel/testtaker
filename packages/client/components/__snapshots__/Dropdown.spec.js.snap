// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Dropdown matches snapshot 1`] = `
<div
  className="dropdown-container"
>
  <select
    className="dropdown"
    name="Lorem Ipsum"
    onChange={[MockFunction]}
  >
    <option
      key="Latin"
      value="Latin"
    >
      Latin
    </option>
  </select>
</div>
`;
